drivers/audio/mp34dt05.c

```diff
@@ -0,0 +1,210 @@
+		if (mp34dt05_private.state == DMIC_STATE_ACTIVE) {
+			tmp_state = DMIC_STATE_CONFIGURED;
+			i2s_cmd = I2S_TRIGGER_STOP;
+		} else {
+			return 0;
+		}
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	ret = i2s_trigger(dev_cfg->i2s_device, I2S_DIR_RX, i2s_cmd);
+	if (ret != 0) {
+		LOG_ERR("trigger failed with %d error", ret);
+		return ret;
+	}
+
+	mp34dt05_private.state = tmp_state;
+	return 0;
+}
+
+static int mp34dt05_configure(struct device *dev, struct dmic_cfg *cfg)
+{
+	int ret;
+	struct mp34dt05_driver_config *const dev_cfg = DEV_CFG(dev);
+	u8_t chan_size = cfg->streams->pcm_width;
+	u32_t audio_freq = cfg->streams->pcm_rate;
+	TPDMFilter_InitStruct *pdm_filter = &mp34dt05_private.pdm_filter;
+	u16_t factor;
+
+	/* PCM buffer size */
+	mp34dt05_private.pcm_mem_slab = cfg->streams->mem_slab;
+	mp34dt05_private.pcm_mem_size = cfg->streams->block_size;
+
+	/* check requested min pdm frequency */
+	if (cfg->io.min_pdm_clk_freq < MP34DT05_MIN_PDM_FREQ ||
+	    cfg->io.min_pdm_clk_freq > cfg->io.max_pdm_clk_freq) {
+		return -EINVAL;
+	}
+
+	/* check requested max pdm frequency */
+	if (cfg->io.max_pdm_clk_freq > MP34DT05_MAX_PDM_FREQ ||
+	    cfg->io.max_pdm_clk_freq < cfg->io.min_pdm_clk_freq) {
+		return -EINVAL;
+	}
+
+	/* calculate oversampling factor based on pdm clock */
+	for (factor = 64; factor <= 128; factor += 64) {
+		u32_t pdm_bit_clk = (audio_freq * factor * \
+				     cfg->channel.req_num_chan);
+
+		if (pdm_bit_clk >= cfg->io.min_pdm_clk_freq &&
+		    pdm_bit_clk <= cfg->io.max_pdm_clk_freq) {
+			break;
+		}
+	}
+
+	if (factor != 64 && factor != 128) {
+		return -EINVAL;
+	}
+
+	/* init the filter lib */
+	pdm_filter->LP_HZ = audio_freq / 2;
+	pdm_filter->HP_HZ = 10;
+	pdm_filter->Fs = audio_freq;
+	pdm_filter->Out_MicChannels = 1;
+	pdm_filter->In_MicChannels = 1;
+	pdm_filter->Decimation = factor;
+	pdm_filter->MaxVolume = 64;
+
+	Open_PDM_Filter_Init(pdm_filter);
+
+	/* configure I2S channels */
+	struct i2s_config i2s_cfg;
+
+	i2s_cfg.word_size = chan_size;
+	i2s_cfg.channels = cfg->channel.req_num_chan;
+	i2s_cfg.format = I2S_FMT_DATA_FORMAT_LEFT_JUSTIFIED |
+			 I2S_FMT_BIT_CLK_INV;
+	i2s_cfg.options = I2S_OPT_FRAME_CLK_MASTER | I2S_OPT_BIT_CLK_MASTER;
+	i2s_cfg.frame_clk_freq = audio_freq * factor / chan_size;
+	i2s_cfg.block_size = mp34dt05_private.pcm_mem_size * (factor / chan_size);
+	i2s_cfg.mem_slab = &rx_pdm_mem_slab;
+	i2s_cfg.timeout = 2000;
+
+	ret = i2s_configure(dev_cfg->i2s_device, I2S_DIR_RX, &i2s_cfg);
+	if (ret != 0) {
+		LOG_ERR("I2S device configuration error");
+		return ret;
+	}
+
+	mp34dt05_private.state = DMIC_STATE_CONFIGURED;
+	return 0;
+}
+
+static const struct _dmic_ops mp34dt05_driver_api = {
+	.configure		= mp34dt05_configure,
+	.trigger		= mp34dt05_trigger,
+	.read			= mp34dt05_read,
+};
+
+static int mp34dt05_initialize(struct device *dev)
+{
+	struct mp34dt05_driver_config *const dev_cfg = DEV_CFG(dev);
+
+	/* bind I2S */
+	dev_cfg->i2s_device = device_get_binding(DT_I2S_5_NAME);
```

Comment:
You should not use directly "DT_I2S_5_NAME" here.
You should make this components a i2s-device on the model of spi-device and then here use generated DT_XXXX_BUS_NAME

Reply:
Actually I do not see for I2S a BUS_NAME property as there is for SPI. I only see:

#define DT_I2S_5_BASE_ADDRESS      DT_ST_STM32_I2S_40015000_BASE_ADDRESS                                                              
#define DT_I2S_5_IRQ_PRI                    DT_ST_STM32_I2S_40015000_IRQ_0_PRIORITY
#define DT_I2S_5_NAME                        DT_ST_STM32_I2S_40015000_LABEL
#define DT_I2S_5_IRQ                            DT_ST_STM32_I2S_40015000_IRQ_0
#define DT_I2S_5_CLOCK_BITS            DT_ST_STM32_I2S_40015000_CLOCK_BITS
#define DT_I2S_5_CLOCK_BUS             DT_ST_STM32_I2S_40015000_CLOCK_BUS

Is it possible that this is due to the fact that I2S dts, differently from SPI, does not use #address-cells and #size-cells clause? See @galak comments on #9034.

Reply:
We're actually looking for a DT_XXXX_MASTER_DEV_NAME, cf other comment